projects/js-packages/publicize-components/src/components/share-status/share-status-action.tsx

```diff
@@ -1,26 +1,71 @@
-import { ExternalLink } from '@wordpress/components';
+import { Button, ExternalLink, Tooltip } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as editorStore } from '@wordpress/editor';
 import { __ } from '@wordpress/i18n';
+import { Icon, linkOff } from '@wordpress/icons';
+import { useCallback } from 'react';
+import useSharePost from '../../hooks/use-share-post';
+import { store as socialStore } from '../../social-store';
 import styles from './styles.module.scss';
 
+type ShareStatusActionProps = {
+	status: string;
+	shareLink: string;
+	connectionId: number;
+};
+
 /**
  *
  * Share status action component.
  *
- * @param {object}  props           - component props
- * @param {boolean} props.status    - status of the share
- * @param {string}  props.shareLink - link to the share
+ * @param {ShareStatusActionProps} props - component props
  * @return {import('react').ReactNode} - React element
  */
-export function ShareStatusAction( { status, shareLink } ) {
-	return (
-		<div className={ styles[ 'share-status-action-wrapper' ] }>
-			{ 'success' !== status ? (
-				<span>Retry</span>
-			) : (
+export function ShareStatusAction( { connectionId, status, shareLink }: ShareStatusActionProps ) {
+	// @ts-expect-error -- `@wordpress/editor` is badly typed, causes issue in CI
+	const postId = useSelect( select => select( editorStore ).getCurrentPostId(), [] );
+	const connections = useSelect( select => select( socialStore ).getConnections(), [] );
+
+	const { doPublicize } = useSharePost( postId );
+
+	const onRetry = useCallback( () => {
+		const skippedConnections = connections.filter(
+			connection => connection.connection_id !== connectionId.toString()
+		);
+
+		// This means that the connection that failed is not in the list of connections anymore.
+		if ( skippedConnections.length === connections.length ) {
```

Comment:
I suppose this means the connection being shared to isn't connected anymore. But is this the best way to determine it? Wouldn't relying on connections  from `useSelect( select => select( socialStore ).getConnections(), [] );` be better?

Also, if the option to reshare is disabled/not available for deleted connections, does it make sense to have this check? If not, we can have this line removed.

Reply:
> I suppose this means the connection being shared to isn't connected anymore. But is this the best way to determine it? Wouldn't relying on connections from `useSelect( select => select( socialStore ).getConnections(), [] );` be better?

It already uses `select( socialStore ).getConnections()` above.

> Also, if the option to reshare is disabled/not available for deleted connections, does it make sense to have this check? If not, we can have this line removed.
We should still have this check here because the button can be enabled by DOM manipulation.

Reply:
Yeah, the connections variable is already coming from the social store; we compare that with the current one.

Reply:
Shall we simplify it further? Pseudo-code below

```
if ( ! connections.includes( connectionId ) ) {

 return;
}

skippedConnectionIds = connections.filter ( connection => connection.connection_id !== connectionId.toString()).map( connection => connection.id ) )

```

Reply:
For that first if, we have to use Array.find, because connections are objects, which adds another iteration on the connection object, increasing the onRetry function's complexity. I tend to lean towards keeping the current solution because of that.

Reply:
`skippedConnections.length === connections.length` will be `true` if there are no connections.

Reply:
We can simply do this

```ts
const onRetry = useCallback( () => {
	const connectionStillExists = connections.some(
		connection => connection.connection_id === connectionId.toString()
	);

	if ( ! connectionStillExists ) {
		return;
	}

	doPublicize( [ connectionId ] );
}, [ connectionId, connections, doPublicize ] );
```

Reply:
> skippedConnections.length === connections.length will be true if there are no connections.

Why is that a problem? We can only retry if there is a connection which you want to retry to

Reply:
That solution is incorrect, because we need to pass in the skipped connection Id's not the one that we share to

Reply:
Ah yes. Thanks. I think your solution is perfect. Let us stick with that.